agent/src/util/config_action.rs

```diff
@@ -548,4 +584,64 @@ mod config_action_tests {
 
         futures::future::join_all(vec![first_add_handle, second_add_handle]).await;
     }
+
+    // Tests that when a Configuration is updated,
+    // if generation has changed, should return true
+    #[tokio::test]
+    async fn test_should_recreate_config_new_generation() {
+        let path_to_config = "../test/yaml/config-a.yaml";
+        let config_yaml = fs::read_to_string(path_to_config).expect("Unable to read file");
+        let mut config: Configuration = serde_yaml::from_str(&config_yaml).unwrap();
+
+        let (stop_discovery_sender, _) = broadcast::channel(2);
+        let (_, finished_discovery_receiver) = mpsc::channel(2);
+
+        let config_info = ConfigInfo {
+            instance_map: Arc::new(Mutex::new(HashMap::new())),
+            stop_discovery_sender: stop_discovery_sender.clone(),
+            finished_discovery_receiver,
+            last_generation: Some(1),
+        };
+        let config_name = config.metadata.name.clone().unwrap();
+        let config_map: ConfigMap = Arc::new(Mutex::new(HashMap::new()));
+        config_map.lock().await.insert(config_name, config_info);
```

Comment:
Nit: Can the shared portion of these two tests be moved to a helper function that both tests call to reuse the code?

Reply:
Yep ... working on that now. Not sure if its just me, but it takes a while to run the tests, build, etc. I suspect it has do to with the fact that I'm using a vscode devcontainer, which runs everything in docker BUT the source files are mounted into the container. And container mounts on Docker Desktop on Mac (and presumably windows) are notoriously slow. Ultimately it's not docker's fault...its the VM's shared folder implementation that is slow.

Reply:
Cargo can certainly take a fair bit of time sometimes as the dependencies directory can really blow up. Sometimes it helps to just run and build the pertinent parts: `cargo build -p agent` or `cargo test <test name>`.

Reply:
Gotcha...is there a way to have the `cargo test ...` command only build the agent stuff? It seems like it spends most of the time building everything else.

Reply:
ah...nvm..its same as build. https://doc.rust-lang.org/cargo/commands/cargo-test.html#package-selection

Reply:
ideally the order wouldnt matter!